### PR TITLE
IAST multipart request endpoint and Java test

### PR DESCRIPTION
**Pull request by DDJavierSantos**
## Description

New multipart parameter test and Java endpoint

JIRA: [APPSEC-11184](https://datadoghq.atlassian.net/browse/APPSEC-11184)

## Motivation

To test tainting of parameters in multipart requests

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

[APPSEC-11184]: https://datadoghq.atlassian.net/browse/APPSEC-11184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: manifests/dotnet.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ tests/:
           TestInsecureCookie: missing_feature
         test_ldap_injection.py:
           TestLDAPInjection: missing_feature
+        test_multpart.py:
```
Comment (cbeauchesne) [inline manifests/dotnet.yml:27]: Typo in all manifests: 

```suggestion
        test_multipart.py:
```

File: manifests/java.yml [hunk 2]
```diff
@@ -54,6 +54,10 @@ tests/:
             spring-boot-3-native: missing_feature (GraalVM. Tracing support only)
             vertx3: v1.12.0
             vertx4: v1.12.0
+        test_multpart.py:
+          TestMultipart:
+            "*": "missing_feature"
+            spring-boot: "v1.22.0"
```
Comment (cbeauchesne) [inline manifests/java.yml:60]:
```suggestion
            spring-boot: v1.22.0
```

File: manifests/java.yml [hunk 3]
```diff
@@ -54,6 +54,10 @@ tests/:
             spring-boot-3-native: missing_feature (GraalVM. Tracing support only)
             vertx3: v1.12.0
             vertx4: v1.12.0
+        test_multpart.py:
+          TestMultipart:
+            "*": "missing_feature"
```
Comment (cbeauchesne) [inline manifests/java.yml:59]:
```suggestion
            '*': missing_feature
```

File: manifests/java.yml [hunk 4]
```diff
@@ -126,7 +130,7 @@ tests/:
           Test_XContentSniffing:
             "*": "missing_feature"
             sping-boot-undertow: "bug (APPSEC-11353)"
-            spring-boot: "bug (APPSEC-11353)"
+            spring-boot: "v1.20.0"
```
Comment (cbeauchesne) [inline manifests/java.yml:133]:
```suggestion
            spring-boot: v1.20.0
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: manifests/dotnet.yml (hunk 1 above)
Comment (cbeauchesne) [reply manifests/dotnet.yml]: The typo is also on other manifest files.

**Review by cbeauchesne — COMMENTED** [review]
CI is failing, I set the PR as draft. Yo ucan set it back to normal once it's ready.

File: tests/appsec/iast/source/test_multipart.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    source_fixture = SourceFixture(
+        http_method="POST",
+        endpoint="/iast/source/multipart/test",
+        request_kwargs={"files": {"file1": ("file1", "bsldhkuqwgervf")}},
+        source_type="http.request.multipart.parameter",
+        source_name="Content-Disposition",
+        source_value=None,
+    )
+
+    @bug(weblog_variant="jersey-grizzly2", reason="Not reported")
```
[the remark this answers is not in the document]
Comment (DDJavierSantos) [reply tests/appsec/iast/source/test_multipart.py]: removed

File: manifests/java.yml [hunk 6]
```diff
@@ -126,7 +126,7 @@ tests/:
           Test_XContentSniffing:
             "*": "missing_feature"
             sping-boot-undertow: "bug (APPSEC-11353)"
-            spring-boot: "bug (APPSEC-11353)"
+            spring-boot: v1.20.0
```
Comment (smola) [inline manifests/java.yml:129]: I think this line should not be touched in this PR, X-Content-Type-Options issues are better addressed on their own.

File: manifests/java.yml [hunk 7]
```diff
@@ -195,6 +195,10 @@ tests/:
             spring-boot-3-native: missing_feature (GraalVM. Tracing support only)
             vertx3: v1.12.0
             vertx4: v1.12.0
+        test_multipart.py:
+          TestMultipart:
+            '*': missing_feature
+            spring-boot: v1.22.0
```
Comment (smola) [inline manifests/java.yml:201]: Let's add all spring-boot variants.

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIastSource.java [hunk 8]
```diff
@@ -104,4 +110,14 @@ private String find(final Collection<String> list,
         return find(list, matcher, Function.identity());
     }
 
+    @PostMapping("/multipart/test")
+    public String handleFileUpload(@RequestParam("file1") MultipartFile file) {
+        String fileContent = "NO_FILE";
+        try {
+            fileContent = Arrays.toString(file.getBytes());
+            file.getOriginalFilename();
```
Comment (smola) [inline utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIastSource.java:118]: For the vulnerability to be trigger, you should use the filename in a sink, like SQL injection. See the other endpoints above.

**Review by smola — CHANGES_REQUESTED** [review]

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIastSource.java (hunk 8 above)
Comment (DDJavierSantos) [reply utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIastSource.java]: done

File: manifests/java.yml (hunk 7 above)
Comment (DDJavierSantos) [reply manifests/java.yml:201]: added

File: manifests/java.yml (hunk 6 above)
Comment (DDJavierSantos) [reply manifests/java.yml]: fixed

File: manifests/java.yml (hunk 7 above)
Comment (DDJavierSantos) [reply manifests/java.yml:201]: done

**Review by smola — APPROVED** [review]

File: docs/weblog/README.md [hunk 9]
```diff
@@ -229,6 +229,9 @@ A GET request using a cookie with name `table` and any value. The value must be
 
 A GET request using a cookie with name `user` and any value. The name must be used in the vulnerability.
 
+#### POST /iast/source/multipart/test
+a POST request
```
Comment (cbeauchesne) [inline docs/weblog/README.md:233]: Could be a little bit better ;)

File: tests/appsec/iast/source/test_multipart.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        request_kwargs={"files": {"file1": ("file1", "bsldhkuqwgervf")}},
+        source_type="http.request.multipart.parameter",
+        source_name="Content-Disposition",
+        source_value=None,
+    )
+
+    def test_source_reported(self):
+        self.source_fixture.test()
+
+    @bug(library="java", reason="Not working as expected")
```
Comment (cbeauchesne) [inline tests/appsec/iast/source/test_multipart.py:25]: Is there a jira ticket for this? if yes, you can simply set the JIRA ticket as `reason`

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIast.java [hunk 11]
```diff
@@ -314,12 +314,14 @@ void secureXSS(final ServletResponse response) throws IOException {
 
     @GetMapping(value = "/hstsmissing/test_insecure", produces = "text/html")
     public String hstsHeaderMissingInsecure(HttpServletResponse response) {
+        response.addHeader("X-Content-Type-Options", "nosniff");
```
Comment (cbeauchesne) [inline utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIast.java:317]: This line will re-introduce the flakyness, no ? see #1666 
cc @manuel-alvarez-alvarez

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: docs/weblog/README.md (hunk 9 above)
Comment (smola) [inline docs/weblog/README.md:233]:
```suggestion
A multipart request uploading a file (with a file name).
```

Comment (DDJavierSantos) [reply docs/weblog/README.md]: The comment got truncated by mistake. Fixed

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIast.java (hunk 11 above)
Comment (DDJavierSantos) [reply utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIast.java:317]: removed

File: tests/appsec/iast/source/test_multipart.py (hunk 10 above)
Comment (smola) [reply tests/appsec/iast/source/test_multipart.py]: It's a common issue across most telemetry tests in this directory, we'll do an overall review some time soon.

Comment (smola) [reply tests/appsec/iast/source/test_multipart.py]: Oh, actually it doesn't apply to this test.

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIast.java (hunk 11 above)
Comment (cbeauchesne) [reply utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/AppSecIast.java:317]: I don't see it removed, and it's failing : https://github.com/DataDog/system-tests/actions/runs/6480588115/job/17596485817?pr=1657

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

**Review by cbeauchesne — APPROVED** [review]
PHP failure are not related, see DataDog/system-tests#1724

**Pull request merged**